ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala

```diff
@@ -296,19 +308,20 @@ final class SandboxServer(config: SandboxConfig) extends AutoCloseable {
 
   }
 
-  private def start()(implicit executionContext: ExecutionContext): Resource[SandboxState] = {
+  private def start(port: Option[Int] = None)(implicit executionContext: ExecutionContext): Unit = {
```

Comment:
Whoops, left that `port` in during debugging. I'll remove it.

Reply:
We need to avoid storing the `Resource` because that's what was causing the memory leak. By flatmapping over resources, we capture the old resources, even if we've already released them. (Perhaps we can catch this by checking if a resource has been released during a `flatMap`; what do you think?)

I'm about to push a change to make this return a `Future[SandboxState]` instead, which preserves the semantics without the memory leak. It was definitely cleaner when this returned.

Reply:
Thanks for the explanation and for looking into this!